### ISSUE TITLE
Paper Mario Sticker Star - get stuck for unknown reason

### ISSUE THREAD

**Issue by wwylele** [citra-emu/citra#1472]
With recent commits, this game gets much more playable with less graphical glitches. 
But when I try to rescue this Toad by hammering the bush away, it just gets stuck for no reason - no log shows up, the screen does not freeze (you can see everything is still moving), but no button  responses. On real console, the player is supposed to go to talk with the Toad, but no button works at this time on Citra. 
![citra issue](https://cloud.githubusercontent.com/assets/4592895/13571074/2ea7a40c-e47b-11e5-813c-2ab8ff940674.PNG)
Talking with other Toads is ok and anything else seems fine.

**Comment by wwylele** [comment]
Here is the log, but I think they are not related to this issue since they came far earlier than when it get stuck
![log](https://cloud.githubusercontent.com/assets/4592895/13571223/2d7de5e0-e47c-11e5-8b83-58849f6a4b3b.PNG)

**Comment by mailwl** [comment]
#1465 fixes it?

**Comment by wwylele** [comment]
@mailwl No. The log changed as JamePeng mentioned in that PR, but this issue is still theire. 
![newlog](https://cloud.githubusercontent.com/assets/4592895/13572024/bdd8874a-e480-11e5-811d-6ddc10495d41.PNG)

Again, these logs (with or without #1465) only show up at the title screen and the opening scene, not in the game play, which seems like a graphical issue. I post these logs here just because there were no new log when it gets stuck. I don't think they are related to this issue.

**Comment by yuriks** [comment]
I've seen a similar thing happen in Project Mirai GX. Game softlocks and accepts no input.

**Comment by wwylele** [comment]
Encountered a similar soft lock in Pokemon Mystery Dungeon - Gates to Infinity, at this point (after choosing pokemon)
![issue2](https://cloud.githubusercontent.com/assets/4592895/13608215/6e8b48f8-e55c-11e5-8ab5-63fd24512c14.PNG)
objects are still moving, without any logs.
On real console, the game shouldn't wait for anything and immediately plays the following cutscene.

**Comment by JohnGodgames** [comment]
Maybe it's related to DSP...

**Comment by wwylele** [comment]
@JohnGodgames  Yeah, I am thinking the same thing. I did a tracing and didn't find any special svc or service call except for DSP calls. At present I think I can't do anything more, but just wait for the DSP implement...

**Comment by MerryMage** [comment]
Re: DSP. I suspect it's due games waiting on `SourceStatus` to update; similar things occur in a number of other games. 

Unimplemented at the moment.

**Comment by wwylele** [comment]
@MerryMage I just rechecked it on real console and found that they plays  sound effects at the problem points, so I think you are right - the game is waiting for audio to synchronize it with video. 
I don't know much about DSP and can't help a lot. Looking forward to the great implement :smile:

**Comment by MerryMage** [comment]
@wwylele Try this: https://github.com/MerryMage/citra/tree/audio-core

I've verified with hardware tests that this behavior should match hardware. No audio output yet, just audio buffer management so it appears to the application that audio is being played.

**Comment by wwylele** [comment]
@MerryMage :confused: does not work. Neither for Mario nor for Pokemon. Tested with branch audio-core bf14f22 .

**Comment by MerryMage** [comment]
Thanks for testsing! I'll take a closer look when I have time.

**Comment by MerryMage** [comment]
Thanks for testsing! I'll take a closer look when I have time.

**Comment by MerryMage** [comment]
Looks like it's unlikely to be the DSP. The audio ARM11 thread is free and playing audio merrily while everything else is hung.

**Comment by JohnGodgames** [comment]
maybe it's related to https://github.com/citra-emu/citra/issues/1484 also full log will be usefull

**Comment by wwylele** [comment]
Here is the [full log](http://pastebin.com/2xxZh9Ps) for Pokemon Mystery Dungeon.
I will update the log for Paper Mario later.

**Comment by wwylele** [comment]
[full log](http://pastebin.com/9UcJeg2x) for Paper Mario. Test with current master+1465+1479
All logs show up before ingame.

**Comment by smartkaartik** [comment]
Ya its true that it accepts no input after choosing the pokemon.
![image](https://cloud.githubusercontent.com/assets/14974622/13695925/03e3370c-e789-11e5-9eb6-76c092825624.png)

It should look like this but it get stuck their - https://www.youtube.com/watch?v=nE0rSb_WRMk

**Comment by butzsan** [comment]
Same problem here, but I'm surprise how well it runs on my linux :) (good speed), good job guys

**Comment by smartkaartik** [comment]
May I ask when will games like pokemon x,y work??

Good job guys....

**Comment by Ryoskn** [comment]
Similar problem in Yo-Kai Watch, game fails to respond to any input, animation keeps going and no log entry related to the bug (previous entries are from the game booting, introduction video and starting menu).

![screenshot 2016-04-04 13 54 47](https://cloud.githubusercontent.com/assets/18261784/14247403/28d81530-fa6e-11e5-8bd6-96b67c27b1cd.png)

[Full Log](http://pastebin.com/yY0VgwzC)

**Comment by smartkaartik** [comment]
It runs well on Linux.
Good job guys!!!!!

**Comment by JayFoxRox** [comment]
This issue is getting hard to follow lately. If it derails some more we'll have to reopen this as a new issue. So with that in mind:

@wwylele: Please create a seperate issue for the Pokemon issue and keep this Paper Mario issue clean (= remove the posts about Pokemon).
@Ryoskn: Please remove your post and create a seperate issue for that aswell
@butzsan / @smartkaartik: Please use the forum or IRC for general appreciation. If a bug has been confirmed before there is no need to post again unless you provide additional information (tests with more versions which change the outcome).

Thanks.

**Comment by wwylele** [comment]
@MerryMage 's audio branch + pipe3 branch fixes this.
